### DISCUSSION TITLE
Add my js code to the project

### DISCUSSION THREAD

**Discussion by meriem1998** [TerriaJS/terriajs#5024] [Q&A]
Hello , so I am totally new to terriaJs I started with cloning the Terriamap project and i got the hang of customizing and charging files into the map .I was previously  working with CesiumJs and would like to take that same code and implement it in terrimap project only I have no idea how or if it's possible to begin with , I just want to create a js file with all the things I want in my map or modify some file if possible (or if there is another way ...).

Here are some code I would like to have in my terria map:

https://sandcastle.cesium.com/?src=3D%20Tiles%20Feature%20Picking.html&label=All 
https://sandcastle.cesium.com/?src=CZML%20Model%20-%20Node%20Transformations.html&label=CZML
https://sandcastle.cesium.com/?src=CZML%20Path.html&label=CZML

**Comment by KeyboardSounds** [comment]
Welcome! 🎉 

Rather than thinking of it in terms of converting code from Cesium to Terria, it might be more useful to look at it in terms of which features you want to be able to use. Terria exposes many of Cesium's features with a different API, and it already supports CZML and 3d tiles feature picking. If you want to add specific new features to Terria, I'm happy to give you some pointers on how to go about implementing them.

You've probably discovered this already, but Terria works a bit differently to Cesium, and adding data is usually a matter of modifying a JSON catalog rather than writing JS. Have a look at our docs [here](https://docs.terria.io/guide/customizing/) which explain how to set up a catalog.
